### DISCUSSION TITLE
EuPI sample app proposal about `eupi-presentor`

### DISCUSSION THREAD

**Discussion by phj0407** [euphony-io/euphony#219] [Ideas]
Hello, I and @nodobi are planning to make EuPI sample app, 'eupi-presentor' , and want repository for it! We will use kotlin and compose.

## App
It will have two features.

### PPT Loader

- load and show PPT 
- We will using pptviewer-android library (https://github.com/itsrts/pptviewer-android)

### PPT controler

- send & receive EuPI for PPT page control

**Comment by designe** [comment]
I'm looking forward to see the presentation through `eupi` :)
Thanks for suggesting proposal! Let me create the issue from this!

**Reply by designe** [reply]
How about checking below link for naming & multi-module?
https://github.com/orgs/euphony-io/discussions/209#discussioncomment-3492027

**Reply by phj0407** [reply]
Thank you for suggestion! I think wave series repository is good idea. For example,...

repo name : wave-ppt
module1 : eupi-ppt-loader
module2 : eupi-ppt-controler

For naming, @nodobi and I will discuss more. 
And we have problem with loading ppt file and linking it😢 First, we was trying to use https://github.com/itsrts/pptviewer-android library, but doesn't work well... Can I get other solution for it?

**Reply by zion830** [reply]
@phj0407 
If you want to use a library that is already made, there are other repositories.
* https://github.com/hanlyjiang/AndroidDocumentViewer
* https://github.com/ZhongXiaoHong/superFileView

Also, MS supports Apache POI for third party applications using Java
* https://poi.apache.org/
* [Example code for reading a ppt file](https://www.baeldung.com/apache-poi-slideshow#1-reading-a-presentation)

**Reply by phj0407** [reply]
Thank you for your help! I think using Apache POI can help us and trying to use it😄

**Reply by nodobi** [reply]
Hi @zion830 
We have some problems, can you help us? 😢

We tried to use Apache POI, but as you know, android doesn't support awt.. 
so we tried to migrate java's Grpahics to android graphics, 
but it didn't work well because there was a problem in converting slides to images.

This is the solution that we folund using Apache POI in java.
```java
  String path = "ppt file path"
  XMLSlideShow ppt = new XMLSlideShow(new FileInputStream(path));
    
  Dimension pgsize = ppt.getPageSize();
  List<XSLFSlide> slide = ppt.getSlides();
  BufferedImage img = null;
  for (int i = 0; i < slide.size(); i++) {
     img = new BufferedImage(pgsize.width, pgsize.height,BufferedImage.TYPE_INT_RGB);
     Graphics2D graphics = img.createGraphics();

     graphics.setPaint(Color.white);
     graphics.fill(new Rectangle2D.Float(0, 0, pgsize.width, pgsize.height));

    // How to migrate draw method to android java without using Graphics2D
     slide.get(i).draw(graphics);
  }
```
The main question is how to change the draw method that takes `Graphics2D` type as a parameter to android's `Graphics`
or are there any other way than converting ppt file to images?

**Reply by designe** [reply]
Hi @phj0407 @nodobi!
https://github.com/qhm123/POI-Android/blob/1565a7f4ed1cc45ddc7c9ba941268d635ed1c138/pptx/src/com/example/poipptxnew/OpenActivity.java#L289
this project is the PPT App porting with Apache POI. I think it will help you.

If it is judged that using the `ppt` library is time consuming, 
`pdf` rather than `ppt` is fine. I think the `pdf` might actually be more useful.

**Reply by nodobi** [reply]
Thanks for your help! @designe 
As you said, `pdf` rather than `ppt` seems to be a very good solution too!
I will discuss with @phj0407 about this solution !😊

**Reply by designe** [reply]
@nodobi @phj0407 
There are so many libraries in Android.
Android officially supports PDF viewer.
### references
- https://developer.android.com/reference/android/graphics/pdf/PdfDocument
- https://pspdfkit.com/getting-started/android/?project=existing-project&package-manager=manual-integration
- https://github.com/barteksc/AndroidPdfViewer
